services/src/main/java/io/grpc/services/HealthCheckingLoadBalancerFactory.java

```diff
@@ -357,23 +361,23 @@ public String toString() {
     private class HcStream extends ClientCall.Listener<HealthCheckResponse> {
       private final ClientCall<HealthCheckRequest, HealthCheckResponse> call;
       private final String callServiceName;
-      private final long callCreationNanos;
       private boolean callHasResponded;
 
       HcStream() {
-        callCreationNanos = time.currentTimeNanos();
         callServiceName = serviceName;
         call = subchannel.asChannel().newCall(HealthGrpc.getWatchMethod(), CallOptions.DEFAULT);
       }
 
       void start() {
+        stopwatch.reset().start();
```

Comment:
Why not keep the original behavior?

Reply:
That's something I am also wondering. Wouldn't this be a better way, although the only invocation of HC rpc starts immediately after creation.

https://github.com/grpc/grpc-java/blob/9373b2ce10eec4b60bd10709c7b32e892dafd226/services/src/main/java/io/grpc/services/HealthCheckingLoadBalancerFactory.java#L325-L326

Reply:
There seems a bug. You should not share the stopwatch across the `HealthCheckingLoadBalancerFactory`. Each `HcStream` should use a new stopwatch, and no need to reset when cancel. I would keep the original behavior.

Reply:
You might want to do something similar to https://github.com/grpc/grpc-java/blob/v1.19.0/netty/src/main/java/io/grpc/netty/NettyClientHandler.java#L105